### ISSUE TITLE
Problem about FSE template to a specific page, post...

### ISSUE THREAD

**Issue by arcadreams** [WordPress/gutenberg#49125]
### Description

FSE

When I assign a template to a specific page, it seems that it does not assign it by the id of the page, but by the url.

So if I change the url to that page, the template is damaged and says it can't find it.

It should not know that that page has changed the url and so the template would have to point to the same page id but with a different url...

### Step-by-step reproduction instructions

1. Create a Page: Page A (page-a)
2. Create a FSE Template for a specific page (select Page A)
3. Go to Page A and change the url (page-a1)
4. The relationship of the template with the page has been broken, and if you go to Template, you see the templates and the one that says Page A / page-a says that it can't find the page
5. I understand that when you change the URL, the relationship should be maintained, it could be associated-perhaps to the unique ID of the page, or when changing the URL and saving the page/post etc, in the save hook, it will check its dependencies with the url previous to pass them to the new url…

### Environment info

Wordpress Versión 6.1.1
GreenshiftVersión: 0.6 FSE

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Andrew-Starr** [comment]
With the way that the template hierarchy works in WordPress, if you have a template `page-a.html` (or `page-a.php` in a non-FSE environment) then a page with the same slug `page-a` will use that template, even when the template is not assigned to that page.

To ensure the page continues to use that template when the URL (slug) is changed, the page template should be assigned to that page so that the page ID is tied to the template.

This is how the template hierarchy is designed to work, and has always been like this going back to classic/php templates, so this is not specific to FSE templates.

**Comment by arcadreams** [comment]
Thanks for your quick response!

About that:
> To ensure the page continues to use that template when the URL (slug) is changed, the page template should be assigned to that page so that the page ID is tied to the template.

After changing the url, how can I assign the template to the "new page"? because I can't go into the template and modify which specific page it points to.
It only lets me delete the template and create a new one by choosing the page with the changed url.

Thanks,
Sergio

**Comment by Andrew-Starr** [comment]
In the Page Editor, go to the panel on the right and under Page > Summary > Template you can select the template that the page should use.

**Comment by arcadreams** [comment]
Yes, but only the general templates appear there, not those that are specified for a single specific page...

Sergio

**Comment by Andrew-Starr** [comment]
Ahh right I think I misunderstood, sorry.

It seems when creating the new template, the process you followed was: Site Editor > Templates > Add New > Page (Displays a single page)

I had wrongly assumed you created a new Custom Template that can be applied to any post/page.
If you intend to change the URL/slug of the page then creating a custom template would be the way to go, as the template can then be selected in in the page editor.

As a specific page template is meant to be locked to that specific page slug, I'm not sure this would be considered an issue, although it would be good to hear some other opinion on this and whether this could somehow be made clearer when initially creating a page template?

**Comment by arcadreams** [comment]
Ok thanks, I don't know if it's a mistake or not either. However, if when creating the specific template for a specific page you could link to the page id instead of the slug, this would not happen, but I don't know what other implications this change might have.

For the moment I will do what you say, I will create a generic template, without being associated with any specific page, and on the page I will choose the template.

Thanks for everything.

Sergio

**Comment by Thelmachido** [comment]
> When I assign a template to a specific page, it seems that it does not assign it by the id of the page, but by the url. So if I change the url to that page, the template is damaged and says it can't find it.

I checked this on my test site custom templates + changing the page URL is working as expected - _the template continues to be assigned to a page even after the URL is changed._ I created the template from the page editor and also site editor, are you able to reproduce the issue @arcadreams mentioned @Andrew-Starr ? 

Gutenberg 15.3.1
WordPress 6.1.1
TT3 theme

**Comment by arcadreams** [comment]
He created a video, tested it in two different installations (but with the same FSE theme, yes...).

https://d.pr/v/tz9tCe

But before you I had written with the video to the developer of the theme, and he replied:
"You can report bugs for Gutenberg to developers of Gutenberg"

Sergio

**Comment by Andrew-Starr** [comment]
> > When I assign a template to a specific page, it seems that it does not assign it by the id of the page, but by the url. So if I change the url to that page, the template is damaged and says it can't find it.
> 
> I checked this on my test site custom templates + changing the page URL is working as expected - _the template continues to be assigned to a page even after the URL is changed._ I created the template from the page editor and also site editor, are you able to reproduce the issue @arcadreams mentioned @Andrew-Starr ?
> 
> Gutenberg 15.3.1 WordPress 6.1.1 TT3 theme

@Thelmachido The issue is with a Page template (Editor > Templates > Add New > Page) i.e. "This template will be used only for the specific item chosen."
With a Custom Template (Editor > Templates > Add New > Custom Template) the assigned template is not affected after changing the page URL, so there is no issue there.

Instead of requesting to make associating the page template with the page ID, I wonder if it may make more sense for the Page Template slug to automagically be changed to match the page slug when the page URL is changed?
